src/payment/strategies/klarna/klarna-payment-strategy.ts

```diff
@@ -121,6 +132,35 @@ export default class KlarnaPaymentStrategy implements PaymentStrategy {
             }));
     }
 
+    private _getUpdateSessionData(billingAddress: BillingAddress, shippingAddress?: Address): KlarnaUpdateSessionParams | {} {
+        if (billingAddress.countryCode === 'US') {
```

Comment:
I wonder, why are we doing this conditionally for non-US countries? Can't we use the same logic for all of them and just send the data at this stage for both US and the rest?

Reply:
We can do that but if we remove this conditional, data from US stores will be override with the same data so I think that is unnecessary

Reply:
My suggestion was to remove this and the code in bcapp that fills in the data for US merchants.

Reply:
Nevertheless with the suggested approach we wouldn't be meting the acceptance criteria of this other User Story https://jira.bigcommerce.com/browse/INT-1402 which explicitly requires to send the PII data at the earliest point (at create session) in US stores. However what we can do is to change this validation to verify only EU countries instead of countries that are different to US

Reply:
Hi @icatalina, we reviewed this approach with Klarna. They said that for non-EU countries, it is better to still send the shoppers personal info on `create_session` because Klarna runs a prescreening on their end that helps to determine which payment methods are available from the beginning.